File: binance4j-core/src/test/java/com/binance4j/core/client/SignTest.java
package com.binance4j.core.client;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
import static org.junit.jupiter.api.Assertions.assertNotNull;

import org.junit.jupiter.api.Test;

import com.binance4j.core.interceptor.AuthenticationInterceptor;

class SignTest {
	@Test
	void testSignature() {
		assertDoesNotThrow(() -> {
			String signature = AuthenticationInterceptor.sign("Hello World!", "keyboard cat");
			assertNotNull(signature);
		});
	}
}

File: binance4j-core/src/test/kotlin/com/binance4j/core/client/SignTest.kt
package com.binance4j.core.client

import com.binance4j.core.interceptor.AuthenticationInterceptor.Companion.sign
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class SignTest {
    @Test
    fun testSignature() {
        Assertions.assertDoesNotThrow {
            val signature = sign("Hello World!", "keyboard cat")
            Assertions.assertNotNull(signature)
        }
    }
}